validation/Validation.cpp

```diff
@@ -0,0 +1,69 @@
+#include <iostream>
+
+#include <Global.test.hpp>
+
+class ValidationTestDefinitions : public ::testing::Test
+{
+   protected:
+    struct TestCase
+    {
+        std::string testName_;
+        std::vector<std::string> compareWith_;
+        std::vector<double> tolerance_;
+    };
+
+    void SetUp() override
+    {
+        // List of validation test scenarios go here (replacement for a "super yaml file", althought this could be added
+        // in the future)
+        const std::vector<TestCase> scenarios_ = {
+            {"SCENARIO_1: Spherical gravity only", {"GMAT", "OREKIT"}, {1.0e-6, 1.0e-6}},
+            {"SCENARIO_2: EGM96 60x60 gravity only", {"GMAT", "OREKIT"}, {1.0e-6, 1.0e-6}},
+            {"SCENARIO_3: Exponential Atmosphere only", {"OREKIT"}, {1.0e-6, 1.0e-6}},
+            {"SCENARIO_4: Constant thrust in track maneuver", {"GMAT"}, {1.0e-6, 1.0e-6}},
+        };
+    }
+};
+
+TEST_F(OpenSpaceToolkit_Astrodynamics, ValidationRunner)
+{
+    for (const auto& scenario_ : scenarios_)
+    {
+        // Run test
+        // testing::internal::CaptureStdout();
+        std::cout << "Running test: " << scenario_.testName_ << std::endl;
+        std::cout << "Compare with: " << scenario_.compareWith_ << std::endl;
+        std::cout << "Tolerance: " << scenario_.tolerance_ << std::endl;
+        // EXPECT_FALSE(testing::internal::GetCapturedStdout().empty());
+
+        // Call mission sequence app to set up the scenario
+        const MissionSequence missionSequence = {scenario_.testName_};
+
+        // Compare with reference
+        for (const auto& index : len(compareWith_))
+        {
+            // Run scenario
+            const std::vector<ComparisonResults> comparisonResults = missionSequence.run(scenario_.compareWith_[index]);
+
+            // Assert on comparison results
+            EXPECT_GT(scenario_.tolerance_[index], comparisonResults[index]);
+        }
+    }
+}
+
+// Alternative approach, validation test definitions in a separate yaml file and parsed in:
+// std::ifstream configFile("validation/ValidationConfig.yaml");
+// YAML::Node config = YAML::Load(configFile);
```

Comment:
This is the future alternative with `yaml` yeah?

Reply:
Correct.